### PR TITLE
Add documentation for Process.Kill(bool entireProcessTree)

### PR DESCRIPTION
**Pull request by wtgodbe**
Adds documentation for method System.Diagnostics.Process.Kill(bool entireProcessTree)

CC @krwq for area co-ownership, @bgribaudo who added this method, @carlossanlop for CoreFx documentation effort ownership, and @rpetrusha @mairaw for grammar concerns.

### CODE REVIEW

**Comment by carlossanlop** [comment]
I think it looks good. If @wtgodbe aggrees with the additional commits, then maybe we can get it merged.

**Comment by mairaw** [comment]
The reflection tool will undo your last change @carlossanlop from what I can tell 😉

**Comment by mairaw** [comment]
@wtgodbe you can see how the changes would look like here: https://review.docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill?view=netcore-3.0&branch=pr-en-us-2828

File: xml/System.Diagnostics/Process.xml [hunk 1]
```diff
@@ -2054,6 +2054,44 @@ There are problems accessing the performance counter API's used to get process i
...
+        <summary>To be added.</summary>
+        <remarks>
+          <format type="text/markdown"><![CDATA[  
+  
+## Remarks  
+
+The `Kill` method forces a termination of the process, while <xref:System.Diagnostics.Process.CloseMainWindow%2A> only requests a termination. 
+When a process with a graphical interface is executing, its message loop is in a wait state. 
+The message loop executes every time a Windows message is sent to the process by the operating system. 
+Calling <xref:System.Diagnostics.Process.CloseMainWindow%2A> sends a request to close to the main window, which, in a well-formed application, closes child windows and revokes all running message loops for the application.
```
Comment (rpetrusha) [inline xml/System.Diagnostics/Process.xml]:
```suggestion
Calling <xref:System.Diagnostics.Process.CloseMainWindow%2A> sends a request to close the main window, which, in a well-formed application, closes child windows and revokes all running message loops for the application. 
```

File: xml/System.Diagnostics/Process.xml [hunk 2]
```diff
@@ -2150,9 +2171,33 @@ There are problems accessing the performance counter API's used to get process i
...
         <remarks>To be added.</remarks>
+        <exception cref="T:System.ComponentModel.Win32Exception">The associated process could not be terminated.  
+  
+ -or-  
+  
+ The process is terminating.  
+  
+ -or-  
+  
+ The associated process is a Win16 executable.</exception>
```
Comment (rpetrusha) [inline xml/System.Diagnostics/Process.xml]: Aren't this and line 2136 legacy conditions -- perhaps they should be deleted? (Are there still Win16 executables running on .NET Framework?)

File: xml/System.Diagnostics/Process.xml [hunk 3]
```diff
@@ -2054,6 +2054,44 @@ There are problems accessing the performance counter API's used to get process i
...
+  
+## Remarks  
+
+The `Kill` method forces a termination of the process, while <xref:System.Diagnostics.Process.CloseMainWindow%2A> only requests a termination. 
+When a process with a graphical interface is executing, its message loop is in a wait state. 
+The message loop executes every time a Windows message is sent to the process by the operating system. 
+Calling <xref:System.Diagnostics.Process.CloseMainWindow%2A> sends a request to close to the main window, which, in a well-formed application, closes child windows and revokes all running message loops for the application. 
+The request to exit the process by calling <xref:System.Diagnostics.Process.CloseMainWindow%2A> does not force the application to quit.
+The application can ask for user verification before quitting, or it can refuse to quit. 
+To force the application to quit, use the `Kill` method.
```
Comment (rpetrusha) [inline xml/System.Diagnostics/Process.xml]:
```suggestion
To force the application to quit, use the `Kill` method.

```

**Comment by bgribaudo** [comment]
Thanks for working on this, @wtgodbe! I added a couple change ideas as comments on code.

**Comment by carlossanlop** [comment]
@bgribaudo do you mean you backported the comments added here into the source code, or did you mean to add suggestions to this PR?

**Comment by wtgodbe** [comment]
Thanks for the feedback @rpetrusha, I've pushed a commit to address it. @bgribaudo I couldn't find your comments, could you point me towards them?

**Comment by bgribaudo** [comment]
@carlossanlop:

> @bgribaudo do you mean you backported the comments added here into the source code, or did you mean to add suggestions to this PR?

Oops! I meant as comments on this change's Markdown code diff. Sorry for the confunsion. :-(

**Comment by bgribaudo** [comment]
@wtgodbe:
> @bgribaudo I couldn't find your comments, could you point me towards them?

Maybe they didn't show up because I hadn't submitted the review. Can you see them now?

**Comment by mairaw** [comment]
Now they appear @bgribaudo!

**Comment by wtgodbe** [comment]
@bgribaudo @carlossanlop @rpetrusha @mairaw PTAL at https://github.com/dotnet/dotnet-api-docs/pull/2828/commits/61b1a6dc107dd0c81f130bef36bca62cb1611245

**Comment by rpetrusha** [comment]
The build failed. I've corrected the error, and a new build has started.

**Comment by carlossanlop** [comment]
@rpetrusha thanks for correcting the error, the build passed. This is fully baked and ready to get merged.

**Comment by rpetrusha** [comment]
Thanks for the additional changes, @wtgodbe. I'll merge your PR now.

**Comment by bgribaudo** [comment]
Thank you, @wtgodbe and team!

**Pull request merged**